src/vmm/src/devices/virtio/device.rs

```diff
@@ -174,10 +174,9 @@ pub trait VirtioDevice: AsAny + Send {
     /// Checks if the resources of this device are activated.
     fn is_activated(&self) -> bool;
 
-    /// Optionally deactivates this device and returns ownership of the guest memory map, interrupt
-    /// event, and queue events.
-    fn reset(&mut self) -> Option<(EventFd, Vec<EventFd>)> {
-        None
+    /// Optionally deactivates this device.
+    fn reset(&mut self) -> Result<(), ResetError> {
+        Err(ResetError::NotImplemented)
```

Comment:
Yes, that's what I had in mind honestly. I _assume_ the initial idea of giving back the `EventFd`s was to "ensure" through the type system that they can't be used, but that is a very dubious choice, IMHO.